src/main/java/net/neoforged/neoforge/common/extensions/IAttributeExtension.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.extensions;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Locale;
+import java.util.function.Consumer;
+import net.minecraft.ChatFormatting;
+import net.minecraft.Util;
+import net.minecraft.network.chat.CommonComponents;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextColor;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.Attribute.Sentiment;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier.Operation;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.common.util.AttributeUtil;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+public interface IAttributeExtension {
+    public static final DecimalFormat FORMAT = Util.make(new DecimalFormat("#.##"), fmt -> fmt.setDecimalFormatSymbols(DecimalFormatSymbols.getInstance(Locale.ROOT)));;
+
+    /**
+     * Converts the value of an attribute modifier to the value that will be displayed.
+     * <p>
+     * For multiplication modifiers, this method is responsible for converting the value to percentage form.<br>
+     * The only vanilla attribute which performs value formatting is Knockback Resistance.<br>
+     *
+     * @param op    The operation of the modifier. Null if we are just displaying the raw value and not a modifier.
+     * @param value The value of the modifier.
+     * @param flag  The tooltip flag.
+     * @return The component form of the formatted value.
+     */
+    default MutableComponent toValueComponent(@Nullable Operation op, double value, TooltipFlag flag) {
+        // Knockback Resistance should be a PercentageAttribute, but we can't registry replace attributes, so we do this here.
+        // For Knockback Resistance, vanilla hardcodes a multiplier of 10 for addition values to hide numbers lower than 1,
+        // but representing the value as a percentage better expresses the intent of the modifier.
+        if (this == Attributes.KNOCKBACK_RESISTANCE.value()) {
+            return Component.translatable("neoforge.value.percent", FORMAT.format(value * 100));
+        }
+
+        // Speed has no metric, so displaying everything as percent works better for the user.
+        // However, Speed also operates in that the default is 0.1, not 1, so we have to special-case it instead of including it above.
```

Comment:
Similar to my comment for knockback resistance, we could also probably do that for move speed if we add an extra optional parameter/field to `PercentageAttribute` for what the base is scaled as (0.1 instead of 1 in this case)